examples-gallery/plot_particle_in_tube.py

```diff
@@ -0,0 +1,513 @@
+"""
+Particle Flight in Tube
+=======================
+
+A particle of mass :math:`m` is moving from a starting point to an ending point,
+subject to a viscous friction force and to a uniform gravitational field.
+The particle must not leave a tube defined by a curve (centerline) in space and a
+radius. At one point during the motion, it must pass through a narrow gate,
+modelled as a circle.
+
+Interesting maybe this:
+( In what follows all components are w.r.t. the inertial frame N.)
+The curve is given as :math:`X(r) = (f(r, params), g(r, params), h(r, params))`,
+where :math:`r` is the parameter of the curve. Let :math:`cut_{param}` be the parameter
+of the curve where the distance of the particle from the curve is closest. Then
+:math:`( \\dfrac{df}{dr}, \\dfrac{dg}{dr}, \\dfrac{dh}{dr} ) |_{r = cut_{param}}`
+is the tangential vector on the curve at the point of closest distance from the
+particle.
+
+So, I form the equation of the plane, which is perpendicular to the curve at the
+point of closest distance, and contains the point of the particle. The intersection
+of the curve and the plane gives the point of closest distance of the curve
+from the particle.
+This leads to a nonlinear equation for :math:`cut_{param}`, which I add to the
+equations of motion by declaring a new state variable :math:`cut_{param}`.
+
+In addition, I introduce a new state variable :math:`dist`, which is the distance
+from the particle from the curve. The reason I do this is so I can bound it to be
+less than the radius of the tube.
+
+The particle must pass through a narrow gate at an intermediate time. I model the
+gate as a circle with its center on the curve.
+As opty presently does not allow inequalities in instance constraints, I
+introduce a new state variable :math:`gate`, and a new specified variable,
+:math:`gate_h`.
+
+**Constants**
+
+- :math:`m` : particle mass, [kg]
+- :math:`c` : viscous friction coefficient of air [Nms]
+- :math:`a1, a2, a3` : parameters of the curve (centerline) [m]
+- :math:`radius` : radius of the tube [m]
+- :math:`max_z` : maximum height of the particle [m]
+
+
+**States**
+
+- :math:`x, y, z` : position of the particle [m]
+- :math:`v_1, v_2, v_3` : speed of particle [m]
+- :math:`dist` : distance of the particle from the curve [m]
+- :math:`cut_{param}` : parameter of the curve where the distance is closest.
```

Comment:
```suggestion
- :math:`\textrm{cut}_\textrm{param}` : parameter of the curve where the distance is closest.
```

Words in latex math should have `\textrm{}` for correct rendering. There are many instances here where it is needed.